fix(DB/creature_template): Classic Dungeon Bosses Damage Output

LukasVolgger opened the pull request:
Increase / decrease damage output based on the official wow beastiary. 

The current damage values ​​are far too low and were only partially raised with the `DamageModifier`. These values ​​should now be adjusted to the official values ​​without `DamageModifier`.

### Team project: 
Thanks to @BarbzYHOOL and @Si1ker for the help. We discussed everything in a team and worked on it for a long time! 
Big thanks especially to @Si1ker who will fix the TBC dungeon bosses! 

It was a lot of work to filter the bosses over several days and adjust the values ​​manually!

<!-- First of all, THANK YOU for your contribution.
 Please fill this template unless your PR is very simple/straightforward.
 Do not forget to have a look at our Pull Request tutorial: http://www.azerothcore.org/wiki/Contribute#how-to-create-a-pull-request
-->

<!-- WRITE A RELEVANT TITLE -->

## CHANGES PROPOSED:
-  Update `mindmg` and `maxdmg` based on the official wow beastiary in `creature_template`
-  Update `minrangedmg` and `maxrangedmg` based on the factor of increasing / decreasing `mindmg` and` maxdmg` in `creature_template`
- Update `DamageModifier` for all creatures to `1` in `creature_template`

#### In addition, the creatures that are not in the instance itself but in the entrance and are listed in Atlas Loot Enhanced have been adjusted: 

- Brainwashed Noble (596)
- Marisa du'Paige (599)
- Foreman Thistlenettle (626)
- Namdo Bizzfizzle (2683)
- Techbot (6231)

#### In addition, the creatures that are not boss at all but have been listed in Atlas Loot Enhanced have been adjusted:

- Scarlet Trainee (6575)

## ISSUES ADDRESSED:
- Closes 
<!-- If the issue doesn't exist, describe it and how to reproduce it, please. If the issue already exists, just paste the link to the issue you close, like this: Closes https://github.com/azerothcore/azerothcore-wotlk/issues/967 -->


## TESTS PERFORMED:
- Imported SQL script on Win 10 Pro x64
- Checked a few bosses in the game and checked the dmg output
<!-- Does it build without errors? Did you test in-game? What did you test? Did you do all these tests on Linux, Mac or Windows? Other tests performed -->


## HOW TO TEST THE CHANGES:
<!-- We need to confirm the changes first, so try to make the work easy for testers (who are not necessarily coders), please:
 - Which commands to use? Which NPC to teleport to?
 - Do we need to enable debug flags on Cmake?
 - Do we need to look at the console? etc...
 - Other steps
-->
**Before you import the SQL script:** Test a few bosses and note the damage output!
- Import the SQL script and check that no error occurs
- Compare the damage output from before with after

## KNOWN ISSUES AND TODO LIST:
<!-- This is a TODO list with checkboxes to tick -->

#### Not all bosses are listed in the beastiary. I adjusted the values based on this src: https://classicdb.ch/

-  Brainwashed Noble (596)
-  Fel Steed (3864)
-  Shadow Charger (3865)
-  Kam Deepfury (1666)
-  Namdo Bizzfizzle (2683)
-  Crowd Pummeler 9-60 (6229)
-  Fallen Champion (6488)
-  Baelog (6906)
-  Galgann Firehammer (7291)
-  Dustwraith (10081)
-  Avatar of Hakkar (8443) 
-  Hurley Blackbreath (9537)
-  Jed Runewatcher (10509)
-  The Ravenian (10507)
-  Baron Rivendare (10440)
-  Black Guard Swordsmith (11121)
-  Prince Tortheldrin (11486)
-  Knot Thimblejack (14338)

## Target branch(es):
- [x] Master


<!-- NOTE: You do not need to squash your commits, on merge we will squash them for you (when there are too many commits we merge them into one big commit for a cleaner and easy to read history). -->

<!-- NOTE2: If you intend to contribute more than once, you should really join us on our discord channel!
 The link is on our site http://azerothcore.org/ We set cosmetic ranks for our contributors and may give access to special resources/knowledge to them! -->


<!-- Do not remove the instructions below about testing, they will help users to test your PR -->
 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

FrancescoBorzi commented:
excellent job @Volgger !

BarbzYHOOL commented:
Sadly, all these SQL comments might be lost when we reach a new azerothcore version/release

Also it's hard to find them given the nature of our updates names.


Now about the bosses, shouldn't we add the "rank" boss to all of them? Kel'thuzad in naxxramas has the boss rank (many lvl 80 raid bosses have the rank boss)
I can add it to all the lines easily if you don't know how

BarbzYHOOL commented:
hmm for the bosses ranks I suppose you want to do this in another PR because you included non-bosses, I get it

Volgger commented:
Yes the comments are very helpful and should be saved somewhere. 
As I have suggested to you before, I would like to have an AC tools repo where we can easily drop and find such things again.  No matter whether it is SQL Selects or Core scripts.  

As for the ranks, I primarily wanted to do the dmg output and then talk about it in a new  PR because there were still a few open issues, such as evowow values.  Of course, these can be wrong, but they are always correct compared to our current ones.

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -0,0 +1,868 @@
+-- Skum
+UPDATE `creature_template` SET `mindmg` = 59, `maxdmg` = 78, `DamageModifier` = 1 WHERE `entry` = 3674;
+-- Verdan the Everliving
+UPDATE `creature_template` SET `mindmg` = 277, `maxdmg` = 369, `DamageModifier` = 1 WHERE `entry` = 5775;
```
gonna surprise players used to his RIDICULOUS damage previously xD

BarbzYHOOL commented:
If someone has an idea on how to check if a mob has ranged attack (using Ranged DMG columns) or not, then we could also remove all ranged dmg for these mobs as it's useless

Maurowin commented:
Maybe, u can add a new column in the DB to filter next time all bosses more easier?

BarbzYHOOL commented:
@Maurowin can you test this plz? I'll test tomorrow (normally), I will notably test wailing caverns and blackrock depths as I tested them recently

Si1ker commented:
Tested: SQL files included, no problems found.
Tested on some creatures, working with values exposed 👍 

This is what you would expect:

Update: -- Verdan the Everliving
UPDATE `creature_template` SET `mindmg` = 277, `maxdmg` = 369, `DamageModifier` = 1 WHERE `entry` = 5775;

- - Before
![before](https://user-images.githubusercontent.com/61223313/76186792-1b0dc800-6199-11ea-8d2f-369e3a8e460e.png)


- - After
![after](https://user-images.githubusercontent.com/61223313/76186803-2234d600-6199-11ea-979c-54909c3263b1.png)

I have to hurry up with the TBC part 🐱

Volgger commented:
Thx again @Si1ker for testing, nice job! Thanks again @ Si1ker for testing, good job! Don't worry and stress yourself - it should still be fun!

FrancescoBorzi commented:
@Si1ker thanks for your feedback!

PS: I have updated your comment, never forget to put the triple apex in morder to better display code:

![image](https://user-images.githubusercontent.com/75517/76260366-f0367900-6257-11ea-83d5-ac751ec1e752.png)

BarbzYHOOL commented:
tested in blackrock depths and on 2 wailing caverns bosses and it's good !

These mobs will still have to be updated with a boss rank and immunity flags in upcoming PRs

BarbzYHOOL commented:
@Yehonal @Kaev @Rochet2 @FrancescoBorzi @P-kito or other devs, 

did we do the right thing by setting the DamageModifier to 1 after changing the mindamage and maxdamage??

From our ingame tests, damagemodifier affects nothing. Maybe it affects attackpower but we do'nt know what attackpower does.

Please, give us some insight, we don't want to introduce regressions

FrancescoBorzi commented:
@BarbzYHOOL I don't know

BarbzYHOOL commented:
> @BarbzYHOOL I don't know

help bru

P-Kito commented:
I did not dive into this PR so much, but DamageModifier is used in core of course. I am just not sure whether it was used or not by Developers. I remember that "exp" does modify it all, tho.
Joschiwald reworked a lot of it, I can ask him maybe. He has been a Trinitycore developer and Guru for a long time now :-P

It's been too long ago, that I touched the 335 stat system. It has changed so much for later expansions 🔨 

![code](https://i.imgur.com/yrOwfQE.png)

P-Kito commented:
I talked to Joschi about DamageModifier, he was unsure aswell, since it was a long time ago. 
He was the one reworking it alot of TC.

What he basically said:

> 
> min/max should not be modified, its the base damage, it has something to do with sniffs 
> and similiar npcs with similiar attributes having similiar damage
> 
> Also there is situations where calculation makes a huge difference, for example on dismantling

LukasVolgger commented:
@P-Kito 
Sry i overlooked our post. So if i understand correctly we should not merge this?

P-Kito commented:
I have no idea. I just trasmitted what Joschi thinks about it and he is my trinitycore / wow emu guru :-P

BarbzYHOOL commented:
@Si1ker why remove tags?

@P-Kito hardly convincing

LukasVolgger commented:
WIP because i wan't to complete the missing bosses. I adjust them in % of the others which were inc/decreased by that PR in same dungeon

LukasVolgger commented:
Updated the missing creatures now based on this src: https://classicdb.ch/ 
Simply because they are the best values ​​I could find. Averages from other creatures are nowhere near as blizzlike as this one.

LukasVolgger commented:
Tbh i don't know what's wrong with Travis. I remember we had the same problem recently but nothing is wrong here.

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -6,8 +6,7 @@ INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1583667245426891200');
 -- https://de.classic.wowhead.com/
--- https://wotlk.evowow.com/
--- https://db.rising-gods.de/
+-- https://classicdb.ch/
```
```suggestion
```

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -74,10 +73,10 @@ UPDATE `creature_template` SET `mindmg` = 47, `maxdmg` = 63, `DamageModifier` =
+-- Fel Steed (Not listed in Beastiary) Src: https://classicdb.ch/?npc=3864
+UPDATE `creature_template` SET `mindmg` = 154, `maxdmg` = 183 WHERE `entry` = 3864;
+-- Shadow Charger (Not listed in Beastiary) Src: https://classicdb.ch/?npc=3865
+UPDATE `creature_template` SET `mindmg` = 145, `maxdmg` = 172 WHERE `entry` = 3865;
```
as i told you, these are not bosses

BarbzYHOOL replied:
in fact the shadow charger mob has no spawn ingame wtf (fel steed is a mini boss so it's fine)

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -271,8 +270,8 @@ UPDATE `creature_template` SET `mindmg` = 354, `maxdmg` = 470, `DamageModifier`
--- Dustwraith (Not listed in Beastiary) Src: https://wotlk.evowow.com/?npc=10081 / https://db.rising-gods.de/?npc=10081
-UPDATE `creature_template` SET `mindmg` = 583, `maxdmg` = 774, `DamageModifier` = 1 WHERE `entry` = 10081;
+-- Dustwraith (Not listed in Beastiary) Src: https://classicdb.ch/?npc=10081
+UPDATE `creature_template` SET `mindmg` = 64, `maxdmg` = 82 WHERE `entry` = 10081;
```
something's wrong here, he's a lvl 48 boss, the original values looked high but here he looks like harmless and u forgot the damage modifier

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -851,7 +850,7 @@ UPDATE `creature_template` SET `minrangedmg` = 167, `maxrangedmg` = 277 WHERE `e
 UPDATE `creature_template` SET `minrangedmg` = 489, `maxrangedmg` = 720 WHERE `entry` = 14326;
 UPDATE `creature_template` SET `minrangedmg` = 373, `maxrangedmg` = 553 WHERE `entry` = 14327;
-UPDATE `creature_template` SET `minrangedmg` = 219, `maxrangedmg` = 325 WHERE `entry` = 14338;
+UPDATE `creature_template` SET `minrangedmg` = 89, `maxrangedmg` = 132 WHERE `entry` = 14338;
```
? are you sure? looks like a mistake

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -792,15 +791,15 @@ UPDATE `creature_template` SET `minrangedmg` = 470, `maxrangedmg` = 687 WHERE `e
+UPDATE `creature_template` SET `minrangedmg` = 576, `maxrangedmg` = 848 WHERE `entry` = 10507;
 UPDATE `creature_template` SET `minrangedmg` = 324, `maxrangedmg` = 480 WHERE `entry` = 10508;
-UPDATE `creature_template` SET `minrangedmg` = 523, `maxrangedmg` = 769 WHERE `entry` = 10509;
+UPDATE `creature_template` SET `minrangedmg` = 364, `maxrangedmg` = 535 WHERE `entry` = 10509;
```
downscaled here, normal?

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -773,7 +772,7 @@ UPDATE `creature_template` SET `minrangedmg` = 349, `maxrangedmg` = 514 WHERE `e
 UPDATE `creature_template` SET `minrangedmg` = 563, `maxrangedmg` = 814 WHERE `entry` = 9938;
 UPDATE `creature_template` SET `minrangedmg` = 129, `maxrangedmg` = 189 WHERE `entry` = 10080;
-UPDATE `creature_template` SET `minrangedmg` = 401, `maxrangedmg` = 591 WHERE `entry` = 10081;
+UPDATE `creature_template` SET `minrangedmg` = 44, `maxrangedmg` = 65 WHERE `entry` = 10081;
```
probably a mistake here

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -764,7 +763,7 @@ UPDATE `creature_template` SET `minrangedmg` = 516, `maxrangedmg` = 755 WHERE `e
 UPDATE `creature_template` SET `minrangedmg` = 228, `maxrangedmg` = 335 WHERE `entry` = 9499;
 UPDATE `creature_template` SET `minrangedmg` = 455, `maxrangedmg` = 665 WHERE `entry` = 9502;
-UPDATE `creature_template` SET `minrangedmg` = 488, `maxrangedmg` = 713 WHERE `entry` = 9537;
+UPDATE `creature_template` SET `minrangedmg` = 364, `maxrangedmg` = 532 WHERE `entry` = 9537;
```
downscaled, is it normal?

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -689,16 +688,16 @@ UPDATE `creature_template` SET `minrangedmg` = 191, `maxrangedmg` = 287 WHERE `e
 UPDATE `creature_template` SET `minrangedmg` = 39, `maxrangedmg` = 58 WHERE `entry` = 6168;
 UPDATE `creature_template` SET `minrangedmg` = 64, `maxrangedmg` = 94 WHERE `entry` = 6228;
-UPDATE `creature_template` SET `minrangedmg` = 51, `maxrangedmg` = 76 WHERE `entry` = 6229;
+UPDATE `creature_template` SET `minrangedmg` = 204, `maxrangedmg` = 303 WHERE `entry` = 6229;
```
big change in damage, is it normal?

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -639,8 +638,8 @@ UPDATE `creature_template` SET `minrangedmg` = 55, `maxrangedmg` = 81 WHERE `ent
-UPDATE `creature_template` SET `minrangedmg` = 34, `maxrangedmg` = 49 WHERE `entry` = 3864;
-UPDATE `creature_template` SET `minrangedmg` = 143, `maxrangedmg` = 203 WHERE `entry` = 3865;
+UPDATE `creature_template` SET `minrangedmg` = 108, `maxrangedmg` = 154 WHERE `entry` = 3864;
+UPDATE `creature_template` SET `minrangedmg` = 102, `maxrangedmg` = 145 WHERE `entry` = 3865;
```
downscaled, is it normal?

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -601,8 +600,8 @@ UPDATE `creature_template` SET `mindmg` = 242, `maxdmg` = 320, `DamageModifier`
--- Knot Thimblejack (Not listed in Beastiary) Src: https://wotlk.evowow.com/?npc=14338 / https://db.rising-gods.de/?npc=14338
-UPDATE `creature_template` SET `mindmg` = 321, `maxdmg` = 426, `DamageModifier` = 1 WHERE `entry` = 14338;
+-- Knot Thimblejack (Not listed in Beastiary) Src: https://classicdb.ch/?npc=14338
+UPDATE `creature_template` SET `mindmg` = 130, `maxdmg` = 154 WHERE `entry` = 14338;
```
he's not attackable so we don't give a damn xD

BarbzYHOOL replied:
he's not a boss

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -402,8 +401,8 @@ UPDATE `creature_template` SET `mindmg` = 327, `maxdmg` = 434, `DamageModifier`
--- Hurley Blackbreath (Not listed in Beastiary) Src: https://wotlk.evowow.com/?npc=9537 / https://db.rising-gods.de/?npc=9537
-UPDATE `creature_template` SET `mindmg` = 705, `maxdmg` = 935, `DamageModifier` = 1 WHERE `entry` = 9537; 
+-- Hurley Blackbreath (Not listed in Beastiary) Src: https://classicdb.ch/?npc=9537
+UPDATE `creature_template` SET `mindmg` = 526, `maxdmg` = 626 WHERE `entry` = 9537;
```
downscaled, is it normal? looks okay to me

btw the new damage look more logical than maxdmg 935

LukasVolgger commented:
Sry my script to prepare the sql code was apparently faulty, sad that I didn't notice it.  I revert this commit and edit it again the days.
Thx for review, helpwd a lot.

LukasVolgger commented:
Updated:

- removed 2 non bosses
- only adapted the creatures that differ widely from others in the same instance

I don't want to remove the databases where we got the values ​​from for better traceability. The creature values ​​are now all well together.

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -241,8 +242,8 @@ UPDATE `creature_template` SET `mindmg` = 158, `maxdmg` = 210, `DamageModifier`
--- Galgann Firehammer (Not listed in Beastiary) Src: https://classicdb.ch/?npc=7291
-UPDATE `creature_template` SET `mindmg` = 319, `maxdmg` = 410 WHERE `entry` = 7291;
+-- Galgann Firehammer (Not listed in Beastiary) Src: https://wotlk.evowow.com/?npc=7291 / https://db.rising-gods.de/?npc=7291
+UPDATE `creature_template` SET `mindmg` = 125, `maxdmg` = 166, `DamageModifier` = 1 WHERE `entry` = 7291;
```
this one has quite low damage compared to his the other bosses in the instance, no?

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -270,8 +271,8 @@ UPDATE `creature_template` SET `mindmg` = 354, `maxdmg` = 470, `DamageModifier`
--- Dustwraith (Not listed in Beastiary) Src: https://classicdb.ch/?npc=10081
-UPDATE `creature_template` SET `mindmg` = 64, `maxdmg` = 82 WHERE `entry` = 10081;
+-- Dustwraith (Not listed in Beastiary) Src: https://wotlk.evowow.com/?npc=10081 / https://db.rising-gods.de/?npc=10081
+UPDATE `creature_template` SET `mindmg` = 583, `maxdmg` = 774, `DamageModifier` = 1 WHERE `entry` = 10081;
```
looks good but damn he hits hard o_0

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -486,8 +487,8 @@ UPDATE `creature_template` SET `mindmg` = 624, `maxdmg` = 827, `DamageModifier`
--- The Ravenian (Not listed in Beastiary) Src: https://classicdb.ch/?npc=10507
-UPDATE `creature_template` SET `mindmg` = 832, `maxdmg` = 991 WHERE `entry` = 10507;
+-- The Ravenian (Not listed in Beastiary) Src: https://wotlk.evowow.com/?npc=10507 / https://db.rising-gods.de/?npc=10507
+UPDATE `creature_template` SET `mindmg` = 362, `maxdmg` = 479, `DamageModifier` = 1 WHERE `entry` = 10507;
```
here this one seems like he hits very lightly, especially for a bulky guy like him, don't you think?

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -517,8 +518,8 @@ UPDATE `creature_template` SET `mindmg` = 779, `maxdmg` = 1034, `DamageModifier`
--- Baron Rivendare (Not listed in Beastiary) Src: https://classicdb.ch/?npc=10440 
-UPDATE `creature_template` SET `mindmg` = 925, `maxdmg` = 1103 WHERE `entry` = 10440;
+-- Baron Rivendare (Not listed in Beastiary) Src: https://wotlk.evowow.com/?npc=10440 / https://db.rising-gods.de/?npc=10440 
+UPDATE `creature_template` SET `mindmg` = 351, `maxdmg` = 465, `DamageModifier` = 1 WHERE `entry` = 10440;
```
na here it's probably a mistake, i preferred the previous change, looks way more logical, the end boss should hit almost 1k, maybe 700 but definitely not 465

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -539,8 +540,8 @@ UPDATE `creature_template` SET `mindmg` = 653, `maxdmg` = 865, `DamageModifier`
--- Black Guard Swordsmith (Not listed in Beastiary) Src: https://classicdb.ch/?npc=11121
-UPDATE `creature_template` SET `mindmg` = 878, `maxdmg` = 959 WHERE `entry` = 11121;
+-- Black Guard Swordsmith (Not listed in Beastiary) Src: https://wotlk.evowow.com/?npc=11121 / https://db.rising-gods.de/?npc=11121
+UPDATE `creature_template` SET `mindmg` = 448, `maxdmg` = 603, `DamageModifier` = 1 WHERE `entry` = 11121;
```
same (ok now i comment to know for myself what to edit)

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -571,8 +572,8 @@ UPDATE `creature_template` SET `mindmg` = 755, `maxdmg` = 1000, `DamageModifier`
--- Prince Tortheldrin (Not listed in Beastiary) Src: https://classicdb.ch/?npc=11486
-UPDATE `creature_template` SET `mindmg` = 854, `maxdmg` = 1015 WHERE `entry` = 11486;
+-- Prince Tortheldrin (Not listed in Beastiary) Src: https://wotlk.evowow.com/?npc=11486 / https://db.rising-gods.de/?npc=11486
+UPDATE `creature_template` SET `mindmg` = 395, `maxdmg` = 523, `DamageModifier` = 1 WHERE `entry` = 11486;
```
same

BarbzYHOOL commented on data/sql/updates/pending_db_world/rev_1583667245426891200.sql:
```diff
@@ -600,8 +601,8 @@ UPDATE `creature_template` SET `mindmg` = 242, `maxdmg` = 320, `DamageModifier`
--- Knot Thimblejack (Not listed in Beastiary) Src: https://classicdb.ch/?npc=14338
-UPDATE `creature_template` SET `mindmg` = 130, `maxdmg` = 154 WHERE `entry` = 14338;
+-- Knot Thimblejack (Not listed in Beastiary) Src: https://wotlk.evowow.com/?npc=14338 / https://db.rising-gods.de/?npc=14338
+UPDATE `creature_template` SET `mindmg` = 321, `maxdmg` = 426, `DamageModifier` = 1 WHERE `entry` = 14338;
```
friendly mob :D

BarbzYHOOL commented:
silker found something interesting when disarming or something, i don't remember exactly, and i also would like to change some of the last dmg values in the last commit

Si1ker commented:
Effects like disarming are doing well.
Screams or things that reduce attack power are not working at all = they keep same dmg

Knindzagxg commented:
Values from ch database are super super high and there's no way to be true.
I have check every single mob / boss for Ragefire Chasm and tried to see how it's gonna work but it's ... Not able to survive. Those low level dungeons have a lot of enemies and using those values is overkill. 

https://classicdb.ch/?npc=11320 - That NPC for example have damage from 87 - 103 and since you always pull 2 (Sometimes even more) tank dies in few seconds - Tested with level 16 and lvl 15 gear to make it realistic. Taking that high damage does not give any chance to healer to cast healing spell with time, since at very beginning almost every cast is around 2 seconds and with 2-3 mobs per pull tank will die before getting any heal.

Now, you can see full run of dungeon on link bellow
https://www.youtube.com/watch?v=hmH2kAsHRGM

I have been checking all databases around and if pay attention you will see that every classic / tbc have super high values until you reach wotlk database.

For example -> https://wotlkdb.com/?npc=11320 This one have pretty good values (Same as wotlk.evo) and also have option to check classic or tbc. My guess is that there is not enough information towards classic and tbc since all of them are kinda different (Depends on db host) Wotlk DB's are almost 100% same cause we have most information about it and i would take those for fixing.

When it comes to modifier, we always need to have some sort of "margin error" so we can't use only 1. My tests are currently with 1.5 and seems to be working nice at lower level dungeons, but when i go higher, damage is broken. 

Gonna take a while till i get current number and will post again once i get it.

BarbzYHOOL commented:
yes, classic DB have too high damage + it's vanilla so it's higher

however if the tanks could do it in vanilla, you should be able to do it in wotlk no problem :)

online DBs for wotlk are just trinitycore default values, we don't care about those

and finally, the values taken here are from an official book (99% of the values here)

LukasVolgger commented:
Tbh, i don't want to merge this PR anymore.  I don't like the fact that no one knows exactly how the Damage Modifier works and what it does to change it.  Yes, the values ​​from the Bestiary are "official" values, but I think that it worked just as well before PR and the discovery of Barbz.

I now find that PR causes more problems than it solves. Same like the TBC PRs from Si1ker.

BarbzYHOOL commented:
we just need to analyze the cpp to understand if we're doing the right thing, I didn't because I'm not a cpp dev, but since they all refused to help (rochet, kaev, everybody), I guess we should try to analyze what these do

The damage values we have in this PR are very good, we only have two issues left:

1. should we set dmgmodifier to 1 or not? If we don't, I can change it for all values with some SQL + text manipulation, so NOTHING is lost
1. the bug mentioned by Silker above, about Demoralizing Roar -like spells, asked some guys to test but nobody did
1. And one or two mobs that I'd like to change the dmg value

So all in all, I can (probably) solve all of this ALONE but I'm too busy at the moment. Otherwise, if people want to help, it's better.

Knindzagxg commented:
I already tested Shout and it works.
Simple way to test it:
Take any unused NPC and modify it's level to 80, min / max damage to 1000, attack power to 400 and modifier to 1/2.
Use Warrior or Druid to see damage change while NPC is affected by shout. 

Damage will be lower, cause attack power affect min and max damage as well. On lower levels it's hard to notice since attack power is lower and will mostly reduce damage from spells that increase damage from it.

Same goes towards range damage and attack power, and those need to be check carefully since not a lot NPC's use range damage (Some spells are affected, but in most cases it's just regular auto attack - Example in Deadmines where you have them before docks and they use Shoot until you reach them for melee range)

Now, when it comes to modifier it's pretty simple.
NPC X min damage 200 / max 400 with modifier 2.
Result will be damage between 400 - 800.

@LukasVolgger - Idea to change it's perfect mate. I spent whole night working on those and already sorted first 10 Classic Dungeons for tests. (Not only Boss NPC's but every other as well)

Only problem i found with yours are values. When i used every single for Ragefire and Deadmines and we tried to do it in normal way (5 player) we died in every single pull. Damage from classic DB is super high and i have no idea why is it. Some of the spells to regular damage + X which was crazy. 3 out of 5 players would die in first few seconds. 

Now i linked you above video from official Ragefire run and you will see what i mean.

If i'm correct, i should have correct values and modifier today and will share all info today.

Si1ker commented:
This PR should not be merged or reopened as the work done does not resolve the issue.

BarbzYHOOL commented:
this PR will be reopened by me for sure
